src/neuroconv/tools/yaml_conversion_specification/_yaml_conversion_specification.py

```diff
@@ -125,18 +125,27 @@ def run_conversion_from_yaml(
             )
     # To properly mimic a true dandi organization, the full directory must be populated with NWBFiles.
     all_nwbfile_paths = [nwbfile_path for nwbfile_path in output_folder_path.iterdir() if nwbfile_path.suffix == ".nwb"]
-    if any(["temp_nwbfile_name_" in nwbfile_path.stem for nwbfile_path in all_nwbfile_paths]):
-        dandi_metadata_list = []
-        for nwbfile_path in all_nwbfile_paths:
-            dandi_metadata = _get_pynwb_metadata(path=nwbfile_path)
-            dandi_metadata.update(path=nwbfile_path)
+    nwbfile_paths_to_set = [
+        nwbfile_path for nwbfile_path in all_nwbfile_paths if "temp_nwbfile_name_" in nwbfile_path.stem
+    ]
+    if any(nwbfile_paths_to_set):
+        dandi_metadata_list = list()
+        for nwbfile_path_to_set in nwbfile_paths_to_set:
+            dandi_metadata = _get_pynwb_metadata(path=nwbfile_path_to_set)
+            if dandi_metadata["subject_id"] is not None:
+                dandi_metadata["subject_id"] = dandi_metadata["subject_id"].replace(" ", "_")
+            dandi_metadata.update(path=nwbfile_path_to_set)
             dandi_metadata_list.append(dandi_metadata)
-        named_dandi_metadata_list = create_unique_filenames_from_metadata(metadata=dandi_metadata_list)
+        dandi_metadata_with_set_paths = create_unique_filenames_from_metadata(metadata=dandi_metadata_list)
 
-        for named_dandi_metadata in named_dandi_metadata_list:
-            if "temp_nwbfile_name_" in named_dandi_metadata["path"].stem:
```

Comment:
Instead of doing filtering of paths that need to be updated here, doing it ahead of time at 128 (reduces cyclomatic complexity too)